jest.config.ts

```diff
@@ -0,0 +1,201 @@
+/**
+ * For a detailed explanation regarding each configuration property, visit:
+ * https://jestjs.io/docs/configuration
+ */
+
+import type {Config} from 'jest';
```

Comment:
strong believer that we should have added this when we needed it and not in advance, but it's already done now so not a huge deal